challenges/systems/challenge.md

```diff
@@ -125,50 +161,85 @@ We are going to compile the program, test it and get back to you.
+the submission, -2 otherwise.
 
-In case of a positive result, we will connect you to our HR team who will collect one-two references and will work out other details. You can start the reference collection process in parallel if you would like to speed up the process.
+In case of a positive result, we will connect you to our HR team who will
+collect one-two references and will work out other details. You can start the
+reference collection process in parallel if you would like to speed up the
+process.
 
 After reference collection, our ops team will send you an offer.
 
-In case of a negative score result, hiring manager will contact you and share a list of the key observations from the team that affected the result.
+In case of a negative score result, hiring manager will contact you and share a
+list of the key observations from the team that affected the result.
 
 ## Code and project ownership
 
-This is a test challenge and we have no intent of using the code you've submitted in production. This is your work, and you are free to do whatever you feel is reasonable with it. In the scenario when you don't pass, you can open source it with any license and use it as a portfolio project.
+This is a test challenge and we have no intent of using the code you've
+submitted in production. This is your work, and you are free to do whatever you
+feel is reasonable with it. In the scenario when you don't pass, you can open
+source it with any license and use it as a portfolio project.
 
 ## Areas of focus
 
 Teleport focuses on networking, infrastructure and security.
 
 These are the areas we will be evaluating in the submission:
 
-* Use consistent coding style. We follow [Go Coding Style](https://github.com/golang/go/wiki/CodeReviewComments) for the Go language. If you are going to use a different language, please pick coding style guidelines and let us know what they are.
-* At the minimum, create tests for authentication, networking, and unhappy scenario.
-* Make sure builds are reproducible. Pick any vendoring/packaging system that will allow us to get consistent build results.
-* Ensure error handling and error reporting is consistent. The system should report clear errors and not crash under non-critical conditions.
-* Avoid concurrency and networking errors. Most of the issues we've seen in production are related to data races, networking error handling or goroutine leaks. We will be looking for those errors in your code.
-* Security. Use strong authentication and simplest, but robust authorization. Set up the strongest transport encryption you can. Test it.
+* Use consistent coding style. We follow
+  [Go Coding Style](https://github.com/golang/go/wiki/CodeReviewComments) for
+  the Go language. If you are going to use a different language, please pick
+  coding style guidelines and let us know what they are.
+* At the minimum, create tests for authentication, networking, and unhappy
```

Comment:
It's not clear to me what "unhappy scenario" means in this context. Is this meaning to say something like "At a minimum, create at least one test for each of authentication, networking, and an error scenario."?